### PR TITLE
Added in a null check in the setter of the Serializer property in Saml2SecurityTokenHandler

### PR DESCRIPTION
**Pull request by mafurman**
Addresses #855.

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs [hunk 1]
```diff
@@ -64,11 +65,19 @@ public Saml2SecurityTokenHandler()
...
         /// Gets or set the <see cref="Saml2Serializer"/> that will be used to read and write a <see cref="Saml2SecurityToken"/>.
         /// </summary>
+        /// <exception cref="ArgumentNullException">'value' is null.</exception>
         public Saml2Serializer Serializer
         {
-            get;
-            set;
-        } = new Saml2Serializer();
+            get { return _serializer; }
+
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs]: remove this line

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs [hunk 2]
```diff
@@ -64,11 +65,19 @@ public Saml2SecurityTokenHandler()
...
+            set
+            {
+                if (value == null)
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs]: use 'null condition operators' pattern, always use LogHelper to throw exceptions.
_serializer = value ?? LogHelper.LogArgumentNullException(nameof(value))

File: test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs [hunk 3]
```diff
@@ -572,11 +572,11 @@ public void CompareSaml2SecurityTokenHandlers()
         {
             TestUtilities.WriteHeader($"{this}.CompareSaml2SecurityTokenHandlers", true);
             var context = new CompareContext($"{this}.CompareSaml2SecurityTokenHandlers");
-            var saml2SecurityTokenHandler1 = new Saml2SecurityTokenHandler { Serializer = null };
+            var saml2SecurityTokenHandler1 = new Saml2SecurityTokenHandler { Serializer = new Saml2Serializer() };
             var saml2SecurityTokenHandler2 = new Saml2SecurityTokenHandler();
             IdentityComparer.AreEqual(saml2SecurityTokenHandler1, saml2SecurityTokenHandler2, context);
 
-            Assert.True(context.Diffs.Count(s => s == "Serializer:") == 1);
+            Assert.True(context.Diffs.Count(s => s == "Serializer:") == 0);
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs]: Do we need this check?

Comment (mafurman) [reply test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs]: No, I don't think we do. I wanted to keep some sort of test for Saml2SecurityTokenHandler, but it doesn't have any other public properties (aside from Serializer) that are checked by IdentityComparer. 

I'll go ahead and remove it entirely.

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs (hunk 1 above)
Comment (mafurman) [reply src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs]: Done! Some properties in other classes put a space between the 'get' and 'set' so I wasn't sure what would be considered the correct style.

**Pull request merged**